codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoderTest.java

```diff
@@ -200,20 +201,6 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
     }
 
     @Test
-    public void dataWriteAfterGoAwayShouldFail() throws Exception {
```

Comment:
@Scottmitch 

I just trust the tests A LOT less, simply due to the fact that they are mocked e.g. that bug with the `ConcurrentModificationException` we had, in a mocked world we would only check if `lifeCycleManager.closeStream(...)` or something gets called but never actually call it and thus not actually get that `Exception`.

Personally, I think the only classes that in the HTTP/2 tests should be mocked is a Channel, ChannelHandlerContext and the FrameWriter. There should not be a need to mock streams, promises, connections, decoders, encoders, lifecyclemanagers and so on ...

Just my 2 cents :-).